bundles/org.openhab.ui/web/src/pages/settings/things/channel/channel-edit.vue

```diff
@@ -55,12 +55,12 @@ export default {
       this.config = Object.assign({}, this.channel.configuration)
       this.$oh.api.get(`/rest/config-descriptions/channel:${this.thing.UID}:${this.channelId.replace('#', '%23')}`).then((ct) => {
         this.configDescription = ct
-        this.ready = true
       }).catch((err) => {
-        if (err === 404) {
+        if (err === 'Not Found' || err === 404) {
           this.noConfig = true
         }
       })
+      this.ready = true
```

Comment:
Isn‘t this wrong? It sets ready when the request is made, not when it is done.

Reply:
I‘ve asked my this question as well, and the answer is not easy:
Whilst I agree that it is set when the request is made, I don‘t think that this really matters. `ready` is only used by the `channel-general-settings` component, which does not need any data that is requested in this component here (all data that component needs is passed to this component and not fetched here). Theoretically, IMO we don‘t need the `ready` variable at all, we could just set the `channel-general-settings`‘s ready prop to true.

What do you think?

Reply:
I used the `ready` flag in my implementations mostly to replace "skeletons" (https://v5.framework7.io/vue/skeleton#examples) with actual data when it was available.

What determines when a page is "ready" to be displayed is up to the developer's choice, there might be additional non-critical requests that might be made after the page is considered "ready".

In that case I'll have to check whether it's a good thing to consider the component "ready" even if the request fails.

Reply:
Let‘s discuss this in my upcoming enhancement PR, and set ready in this one as usual to get this merged and be able to publish the milestone release.

Reply:
I agree.

Reply:
Great, I will change this and merge then.